File: src/main/java/ms/aurora/loader/ClientStub.scala
package ms.aurora.loader

import java.applet.{AppletContext, AppletStub}
import java.net.URL

/**
 * Applet stub provided to the game applet.
 * @author rvbiljouw
 */
class ClientStub(config: ClientConfig) extends AppletStub {
  private var active = false

  def isActive: Boolean = active

  def setActive(active: Boolean) {
    this.active = active
  }

  def getDocumentBase: URL =
    new URL(config.getDocumentBase)

  def getCodeBase: URL =
    new URL(config.getDocumentBase)

  def getParameter(name: String): String =
    config.getParam(name)

  def getAppletContext: AppletContext =
    new ClientContext()

  def appletResize(width: Int, height: Int) {}
}

File: src/main/java/ms/aurora/loader/impl/ClientStub.java
package ms.aurora.loader.impl;

import ms.aurora.loader.ClientConfig;

import java.applet.AppletContext;
import java.applet.AppletStub;
import java.net.MalformedURLException;
import java.net.URL;

/**
 * @author rvbiljouw
 */
public class ClientStub implements AppletStub {
    private final ClientContext context = new ClientContext();
    private final ClientConfig config;

    public ClientStub(ClientConfig config) {
        this.config = config;
    }

    @Override
    public boolean isActive() {
        return true;
    }

    @Override
    public URL getDocumentBase() {
        try {
            return new URL(config.getDocumentBase());
        } catch (MalformedURLException e) {
            e.printStackTrace();
        }
        return null;
    }

    @Override
    public URL getCodeBase() {
        try {
            return new URL(config.getDocumentBase());
        } catch (MalformedURLException e) {
            e.printStackTrace();
        }
        return null;
    }

    @Override
    public String getParameter(String name) {
        return config.getParam(name);
    }

    @Override
    public AppletContext getAppletContext() {
        return context;
    }

    @Override
    public void appletResize(int width, int height) {

    }
}
